include/binaryfusefilter.h

```diff
@@ -858,4 +860,91 @@ static inline bool binary_fuse8_deserialize(binary_fuse8_t * filter, const char
   return true;
 }
 
+// minimal bitfield implementation
+typedef uint8_t bitf_t;
+#define bitf_w (sizeof(bitf_t) * 8)
+#define bitf_sz(bits) (((bits) + bitf_w - 1) / bitf_w)
+#define bitf_word(bit) (bit / bitf_w)
+#define bitf_bit(bit) ((1U << (bit % bitf_w)) % 256)
+
+#define ser(buf, lim, src) do {			\
+	if ((buf) + sizeof src > (lim))		\
+	  return (0);				\
+	memcpy(buf, &src, sizeof src);		\
+	buf += sizeof src;			\
+} while (0)
+
+#define deser(dst, buf, lim) do {		\
+	if ((buf) + sizeof dst > (lim))		\
+	  return (false);			\
+	memcpy(&dst, buf, sizeof dst);		\
+	buf += sizeof dst;			\
+} while (0)
+
+// return required space for binary_fuse8_pack()
+static inline size_t binary_fuse8_pack_bytes(const binary_fuse8_t *filter)
+{
+  size_t sz = 0;
+  sz += sizeof filter->Seed;
+  sz += sizeof filter->Size;
+  sz += bitf_sz(filter->ArrayLength);
+  for (size_t i = 0; i < filter->ArrayLength; i++) {
+    if (filter->Fingerprints[i] == 0)
+      continue;
+    sz += sizeof filter->Fingerprints[i];
+  }
+  return (sz);
+}
+
+// serialize as packed format, return size used or 0 for insufficient space
+static inline size_t binary_fuse8_pack(const binary_fuse8_t *filter, uint8_t *buffer, size_t space) {
+  uint8_t *buf = buffer;
+  uint8_t *e = buf + space;
+
+  ser(buf, e, filter->Seed);
+  ser(buf, e, filter->Size);
+  size_t bsz = bitf_sz(filter->ArrayLength);
+  if (buf + bsz > e)
+    return (false);
```

Comment:
Here you might return (size_t)-1?

Reply:
Thank you, this was an oversight and residue from a previous edit. Will fix as `return (0)` as documented.